libraries/botbuilder-azure-blobs/src/blobsTranscriptStore.ts

```diff
@@ -251,7 +248,7 @@ export class BlobsTranscriptStore implements TranscriptStore {
      * @returns {Promise<void>} A promise representing the async operation.
      */
     async logActivity(activity: Activity): Promise<void> {
-        assertActivity(activity, ['activity']);
+        t.Record({ activity: t.Dictionary(t.Unknown, t.String) }).check({ activity });
```

Comment:
I'm guessing this is the line that directly addresses the issue. Since we know the structure of `Activity`, would it make sense to check the keys and not the values, to ensure we don't get an activity with an invalid key or something (if possible)?

Reply:
Initially, I thought we ought to be quite restrictive in our type assertions. After a few instances of this kind of issue, along with encouragement from @stevengum, I think I'm in favor of more permissive type checks. In this particular scenario, we want to ensure that we have some kind of object that we can log. It doesn't absolutely have to be a perfectly formatted Activity.

I do think we are due for some runtime type checking overhauls which I think we can address when we start working on https://github.com/microsoft/botbuilder-js/issues/3670.